crates/wash-lib/src/cli/par.rs

```diff
@@ -0,0 +1,179 @@
+use std::{fs::File, io::prelude::*, path::PathBuf};
+
+use crate::cli::{extract_keypair, OutputKind};
+use anyhow::{anyhow, bail, Context, Result};
+use nkeys::KeyPairType;
+use provider_archive::ProviderArchive;
+
+const GZIP_MAGIC: [u8; 2] = [0x1f, 0x8b];
+
+pub struct ParCreateArgs {
+    pub capid: String,
+    pub vendor: String,
+    pub revision: Option<i32>,
+    pub version: Option<String>,
+    pub schema: Option<PathBuf>,
+    pub issuer: Option<String>,
+    pub subject: Option<String>,
+    pub name: String,
+    pub directory: Option<PathBuf>,
+    pub arch: String,
+    pub binary: String,
+    pub destination: Option<String>,
+    pub compress: bool,
+    pub disable_keygen: bool,
+}
+pub struct ParInsertArgs {
+    pub archive: String,
+    pub arch: String,
+    pub binary: String,
+    pub directory: Option<PathBuf>,
+    pub issuer: Option<String>,
+    pub subject: Option<String>,
+    pub disable_keygen: bool,
+}
+
+#[allow(clippy::too_many_arguments)]
+pub async fn handle_par_create(
+    ParCreateArgs {
```

Comment:
Since this is in the library I like the idea of calling this something more human-parseable
```suggestion
#[allow(clippy::too_many_arguments)]
pub async fn create_provider_archive(
    ParCreateArgs {
```